mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -429,6 +429,24 @@ PointerInfo AllocTracker::lookupOrDefault(uintptr_t ptr) const {
   return map.at(ptr);
 }
 
+std::pair<int64_t, int64_t> AllocTracker::reportAllocatedMemory() const {
+    int64_t totalCpuMemory = 0;
+    int64_t totalGpuMemory = 0;
+
+    for (const auto &entry : map) {
+        const PointerInfo &info = entry.second;
+        if (info.isExternallyManaged())
```

Comment:
@christopherbate Is this sufficient for tracking only internally managed/allocated pointers?